basic-examples/serial-data_passing/workflow.yaml

```diff
@@ -0,0 +1,24 @@
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties: {}
```

Comment:
Hmmm...given that we're defining no `properties`, it's kind of a shame that we need to define any keys under `objects` or even the `objects` key itself....  (That is, if it is a common case to have a workflow with no inputs, perhaps we should reduce the boilerplate here.)

Reply:
I tried passing a blank object at both the `input` and the `objects` level, but those both failed. This seems to be the simplest option.

Reply:
I'm not entirely surprised that we need to have this much (although, given that the input is empty, it would be cool if we didn't need this, but I guess the engine hangs other stuff off the root object, so we can't completely skip it):
```yaml
input:
  root: RootObject
```
However, that creates a reference to `RootObject` which then needs to be declared, so I suppose we need all of:
```yaml
input:
  root: RootObject
  objects:
    RootObject:
      id: RootObject
```
But, if we're not defining any keys under `properties`, it would be nice to be able to omit that key.

Reply:
I don't think we can reduce the boilerplate much. The advantage of leaving the extra fields is that it becomes easier when you add fields later. But that's obviously negligible.

Reply:
No, being able to omit the `objects` key was wishful thinking, given the requirement for the `root` keyword.

Requiring the `properties` keyword to make things easier later, if you change your mind and decide that having no input wasn't the right choice, sounds silly to me.  I'll grant that it's just a handful of characters on a single line, but providing support for something that an author might never need or want seems like an anti-pattern.